gitlab/resource_gitlab_project_test.go

```diff
@@ -509,6 +509,86 @@ func TestAccGitlabProject_importURL(t *testing.T) {
 	})
 }
 
+func TestAccGitlabProject_importURLMirrored(t *testing.T) {
+	// Since we do some manual setup in this test, we need to handle the test skip first.
+	if os.Getenv(resource.TestEnvVar) == "" {
+		t.Skip(fmt.Sprintf("Acceptance tests skipped unless env '%s' set", resource.TestEnvVar))
+	}
+
+	client := testAccProvider.Meta().(*gitlab.Client)
+	rInt := acctest.RandInt()
+
+	// Create a base project for importing.
+	baseProject, _, err := client.Projects.CreateProject(&gitlab.CreateProjectOptions{
+		Name:       gitlab.String(fmt.Sprintf("base-%d", rInt)),
+		Visibility: gitlab.Visibility(gitlab.PublicVisibility),
+	})
+	if err != nil {
+		t.Fatalf("failed to create base project: %v", err)
+	}
+
+	defer client.Projects.DeleteProject(baseProject.ID)
+
+	// Add a file to the base project, for later verifying the import.
+	_, _, err = client.RepositoryFiles.CreateFile(baseProject.ID, "foo.txt", &gitlab.CreateFileOptions{
+		Branch:        gitlab.String("master"),
+		CommitMessage: gitlab.String("add file"),
+		Content:       gitlab.String(""),
+	})
+	if err != nil {
+		t.Fatalf("failed to commit file to base project: %v", err)
+	}
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectDestroy,
+		Steps: []resource.TestStep{
+			{
+				// First, import, as mirrored
+				Config: testAccGitlabProjectConfigImportURLMirrored(rInt, baseProject.HTTPURLToRepo),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("gitlab_project.imported", "import_url", baseProject.HTTPURLToRepo),
+					resource.TestCheckResourceAttr("gitlab_project.imported", "mirror", "true"),
+					resource.TestCheckResourceAttr("gitlab_project.imported", "mirror_trigger_builds", "true"),
+
+					func(state *terraform.State) error {
+						projectID := state.RootModule().Resources["gitlab_project.imported"].Primary.ID
+
+						_, _, err := client.RepositoryFiles.GetFile(projectID, "foo.txt", &gitlab.GetFileOptions{Ref: gitlab.String("master")}, nil)
+						if err != nil {
+							return fmt.Errorf("failed to get file from imported project: %w", err)
+						}
+
+						return nil
+					},
+				),
+			},
+			{
+				// Second, disable mirroring, using the original ImportURL acceptance test
+				Config: testAccGitlabProjectConfigImportURL(rInt, baseProject.HTTPURLToRepo),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("gitlab_project.imported", "import_url", baseProject.HTTPURLToRepo),
+					resource.TestCheckResourceAttr("gitlab_project.imported", "mirror", "false"),
```

Comment:
@armsnyder is my explicit property check causing the test to fail? When it says `Check 2/4 error`, which check is actually failing?

```
=== RUN   TestAccGitlabProject_importURLMirrored
Error:     testing.go:684: Step 0 error: Check failed: Check 2/4 error: gitlab_project.imported: Attribute 'mirror' expected "true", got "false"
```

Reply:
Trying something, but shooting in the dark: https://github.com/gitlabhq/terraform-provider-gitlab/pull/449/commits/116e25f10739e1ea7f945706eb30fd4542ab4e0a

Reply:
@armsnyder i'm at a loss and i've burnt my whole day on this... the likely solution is to remove the explicit checks as you mentioned earlier, but I'm not seeing in the test cases how the properties would be checked implicitly.

Could you reference where/how implicit checks are made and I'll try to see if that makes the test cases happy?

Reply:
@cnuss,
Looking at the [API documentation](https://docs.gitlab.com/ce/api/projects.html#configure-pull-mirroring-for-a-project), the pull-mirroring is only available in the Starter/Bronze configuration. 

Are you able to make your tests only run if in the Enterprise Edition pipeline?

Reply:
> @cnuss,
> Looking at the [API documentation](https://docs.gitlab.com/ce/api/projects.html#configure-pull-mirroring-for-a-project), the pull-mirroring is only available in the Starter/Bronze configuration.
> 
> Are you able to make your tests only run if in the Enterprise Edition pipeline?

You can skip CE tests by adding `SkipFunc: isRunningInCE` to your test steps.  

Example:
```
{
  // First, import, as mirrored
  SkipFunc: isRunningInCE,
  Config:   testAccGitlabProjectConfigImportURLMirror(rInt, baseProject.HTTPURLToRepo),
  ...
},
```
This means the tests will only run for the EE acceptance tests.

Reply:
@nicholasklick I pushed https://github.com/gitlabhq/terraform-provider-gitlab/pull/449/commits/dee4ce6f2a3425413b2242f3f494ab815604c7ef to include import_url on Project Updates but the build is still failing:

```
Error:     testing.go:684: Step 0 error: Check failed: Check 3/4 error: got mirror false; want true
Ref: https://github.com/gitlabhq/terraform-provider-gitlab/pull/449/checks?check_run_id=1622366823
```

So I assume I have to disable running mirroring in CE, but it appears that EE tests only run for PRs/branches on `gitlabhq/terraform-provider-gitlab`, not a forked repo since the EE token is stored in GH Secrets on the root repo.

I'd like to set `SkipFunc`, could you make a branch for me off of master, and I'll change this PR to merge into that branch instead, then we can merge into a branch owned by `gitlabhq` for EE tests to run? Or, alternatively/additionally can you grant me write access to that branch as well?

Reply:
@cnuss I've setup a branch on `gitlabhq` to run EE tests: https://github.com/gitlabhq/terraform-provider-gitlab/pull/512. Thanks!